engine/src/cmds.h

```diff
@@ -1530,6 +1530,7 @@ class MCOpen : public MCStatement
 	
 	// MM-2014-06-13: [[ Bug 12567 ]] Added new "open socket <socket> with verification for <host>" variant.
 	MCExpression *verifyhostname;
+    MCExpression *fromaddress;
```

Comment:
I suggest using an `MCAutoPointer<MCExpression>` here.  Then it will be automatically initialised and cleaned up, and you can omit the changes to the `MCOpen` constructor and destructor.